Lab06-OOP-Principles/Bird.cs

```diff
@@ -1,13 +1,32 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-
-namespace Lab06_OOP_Principles
-{
-    public abstract class Bird: Animal
-    {
-        public abstract bool Flies { get; }
-    }
-}
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Lab06_OOP_Principles
+{
+    public abstract class Bird: Animal, ILaysEggs, IHasBeak
+    {
+        public abstract bool Flies { get; }
+
+        public bool HasBeak()
+        {
+            Console.WriteLine("Birds have beaks.");
+            return true;
+            //throw new NotImplementedException();
+        }
+
+        public bool LaysEggs()
+        {
+            Console.WriteLine("Birds Lay eggs..");
+            return true;
+            //throw new NotImplementedException();
+        }
+
+        public int NumEggs()
+        {
+            throw new NotImplementedException();
+        }

```

Comment:
Since we don't know how many eggs any given `Bird` might lay, we can implement the interface with an `abstract` method that will be required for subclasses to implement!

```suggestion
        public abstract int NumEggs();
```